sdk/identity/identity/review/identity.api.md

```diff
@@ -109,11 +117,18 @@ export interface DefaultAzureCredentialOptions extends TokenCredentialOptions {
 }
 
 // @public
-export class DeviceCodeCredential implements TokenCredential {
-    constructor(tenantId?: string, clientId?: string, userPromptCallback?: DeviceCodePromptCallback, options?: TokenCredentialOptions);
+export class DeviceCodeCredential implements InteractiveCredential {
+    constructor(tenantId?: string, clientId?: string, options?: DeviceCodeCredentialOptions);
+    constructor(tenantId?: string, clientId?: string, options?: DeviceCodePromptCallback);
+    authenticate(scopes: string | string[], options?: GetTokenOptions): Promise<AuthenticationRecord | undefined>;
```

Comment:
This made writing the tests so much better. Not having to send "undefined" to access the optional parameters feels a lot better to me.